### PR TITLE
Small k8s fixes and optimizations

### PR DESCRIPTION
**Pull request by aanm**
Read per commit basis

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by christarazi — APPROVED** [review]

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml [hunk 1]
```diff
@@ -60,6 +60,9 @@ spec:
             path: /healthz
             port: {{ .Values.global.agent.healthPort }}
             scheme: HTTP
+            httpHeaders:
```
Comment (tklauser) [inline install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:63]: I don't think this is strictly necessary right now, `:9876/healthz` is served by a separate minimal HTTP server which will always only return HTTP 200 or 500. But I guess it's good for forward compatibility, once we might extend this status check API.

**Review by tklauser — APPROVED** [review]

File: install/kubernetes/cilium/charts/agent/templates/daemonset.yaml (hunk 1 above)
Comment (aanm) [reply install/kubernetes/cilium/charts/agent/templates/daemonset.yaml:63]: @tklauser although it is served by a minimal HTTP server there's a significant difference in the CPU consumption and memory usage by the server when setting `brief: true`

**Comment by aanm** [comment]
test-me-please

File: pkg/endpoint/endpoint_status.go [hunk 2]
```diff
@@ -325,7 +325,6 @@ func (e *Endpoint) GetCiliumEndpointStatus(conf EndpointStatusConfiguration) *ci
 		ExternalIdentifiers: modelStatus.ExternalIdentifiers,
 		Identity:            getEndpointIdentity(modelStatus),
 		Networking:          getEndpointNetworking(modelStatus),
-		State:               compressEndpointState(modelStatus.State),
```
Comment (tgraf) [inline pkg/endpoint/endpoint_status.go:328]: This was not the intent. The intent was to publish a state that doesn't change consistently. Why are you making this change? Did you observe many updates? Or are you observing another PATCH required after the initial CREAT?

File: install/kubernetes/cilium/charts/config/templates/configmap.yaml [hunk 3]
```diff
@@ -464,7 +464,7 @@ data:
 
   # A space separated list of iptables chains to disable when installing feeder rules.
   disable-iptables-feeder-rules: {{ .Values.global.disableIptablesFeederRules | join " " | quote }}
-{{ if .Values.global.ipam }}
+{{ if and .Values.global.ipam (not (or .Values.global.gke.enabled ( or .Values.global.eni .Values.global.azure.enabled ))) }}
```
Comment (tgraf) [inline install/kubernetes/cilium/charts/config/templates/configmap.yaml:467]: Why this change? Doesn't this lead to a secondary `ipam:` line? Both Azure and ENI have a separate section which also sets `ipam:`. I'm fine with this but then you have to remove that other one.

File: install/kubernetes/cilium/values.yaml [hunk 4]
```diff
@@ -509,3 +509,6 @@ global:
 
   healthChecking:
     enabled: true
+
+  cnpStatusUpdates:
+    enabled: false
```
Comment (tgraf) [inline install/kubernetes/cilium/values.yaml:514]: This needs a note in the upgrade guide

**Review by tgraf — CHANGES_REQUESTED** [review]

File: pkg/endpoint/endpoint_status.go (hunk 2 above)
Comment (aanm) [reply pkg/endpoint/endpoint_status.go:328]: Reverted

File: install/kubernetes/cilium/charts/config/templates/configmap.yaml (hunk 3 above)
Comment (aanm) [reply install/kubernetes/cilium/charts/config/templates/configmap.yaml:467]: @tgraf I think you missed the "not"? This commit is fixing the "Both Azure and ENI have a separate section which also sets ipam:" since that's the current behavior.

File: install/kubernetes/cilium/values.yaml (hunk 4 above)
Comment (aanm) [reply install/kubernetes/cilium/values.yaml:514]: Please take a look at the doc changes.

File: Documentation/install/upgrade.rst [hunk 5]
```diff
@@ -340,6 +340,13 @@ IMPORTANT: Changes required before upgrading to 1.8.0
   .. note:: The ENI IPAM mode automatically derives the native routing CIDR so
             no action is required.
 
+* To make the usage of Cilium Network Policies (CNP) scalable for clusters with
```
Comment (tgraf) [inline Documentation/install/upgrade.rst]: This doesn't say that we changed this.

I would say something like this:
> The CiliumNetworkPolicy Status includes information which allows to derive when all nodes in a cluster are enforcing a particular CiliumNetworkPolicy. For large clusters running CRD mode, this visibility is costly as it requires all nodes to participate. In order to ensure scalability, CiliumNetworkPolicy status visibility has been disabled for all new deployments. If you want to enable it, set the ConfigMap option ``disable-cnp-status-updates`` to ``false`` by using Helm ``--set global.cnpStatusUpdates.enabled=true`` or by editing the ConfigMap directly.

**Review by tgraf — CHANGES_REQUESTED** [review]

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Review by tgraf — APPROVED** [review]

**Comment by tgraf** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
retest-4.9

**Comment by aanm** [comment]
restest-runtime

**Comment by aanm** [comment]
restest-runtime

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by joestringer** [comment]
Travis hit #11560, will re-kick.

**Comment by joestringer** [comment]
Runtime privileged tests hit this:
https://github.com/cilium/cilium/issues/11624

File: pkg/maps/policymap/policymap_privileged_test.go [hunk 6]
```diff
@@ -101,6 +101,32 @@ func (pm *PolicyMapTestSuite) TestPolicyMapDumpToSlice(c *C) {
...
+func (pm *PolicyMapTestSuite) TestPolicyMapDumpKeysToSlice(c *C) {
+	c.Assert(testMap, NotNil)
+
+	fooEntry := newKey(1, 1, 1, 1)
+	err := testMap.AllowKey(fooEntry, 0)
+	c.Assert(err, IsNil)
+
+	dump, err := testMap.DumpKeysToSlice()
+	c.Assert(err, IsNil)
+	c.Assert(len(dump), Equals, 1)
```
Comment (joestringer) [inline pkg/maps/policymap/policymap_privileged_test.go:117]:
```
[2020-05-20T21:36:16.370Z]   	 === RUN   Test
[2020-05-20T21:36:16.370Z]   	 START: policymap_privileged_test.go:130: PolicyMapTestSuite.TestDeleteNonexistentKey
[2020-05-20T21:36:16.370Z]   	 PASS: policymap_privileged_test.go:130: PolicyMapTestSuite.TestDeleteNonexistentKey	0.000s
[2020-05-20T21:36:16.370Z]   	 
[2020-05-20T21:36:16.370Z]   	 START: policymap_privileged_test.go:104: PolicyMapTestSuite.TestPolicyMapDumpKeysToSlice
[2020-05-20T21:36:16.370Z]   	 PASS: policymap_privileged_test.go:104: PolicyMapTestSuite.TestPolicyMapDumpKeysToSlice	0.000s
[2020-05-20T21:36:16.370Z]   	 
[2020-05-20T21:36:16.370Z]   	 START: policymap_privileged_test.go:78: PolicyMapTestSuite.TestPolicyMapDumpToSlice
[2020-05-20T21:36:16.370Z]   	 policymap_privileged_test.go:87:
[2020-05-20T21:36:16.370Z]   	     c.Assert(len(dump), Equals, 1)
[2020-05-20T21:36:16.370Z]   	 ... obtained int = 2
[2020-05-20T21:36:16.370Z]   	 ... expected int = 1
[2020-05-20T21:36:16.370Z]   	 
[2020-05-20T21:36:16.370Z]   	 FAIL: policymap_privileged_test.go:78: PolicyMapTestSuite.TestPolicyMapDumpToSlice
[2020-05-20T21:36:16.370Z]   	 
[2020-05-20T21:36:16.370Z]   	 OOPS: 2 passed, 1 FAILED
[2020-05-20T21:36:16.370Z]   	 --- FAIL: Test (0.00s)
[2020-05-20T21:36:16.370Z]   	 FAIL
[2020-05-20T21:36:16.370Z]   	 coverage: 5.2% of statements in ./...
[2020-05-20T21:36:16.370Z]   	 FAIL	github.com/cilium/cilium/pkg/maps/policymap	0.077s
[2020-05-20T21:36:16.370Z]   	 FAIL
[2020-05-20T21:36:16.370Z]   	 Makefile:142: recipe for target 'tests-privileged' failed
[2020-05-20T21:36:16.370Z]   	 make: Leaving directory '/home/vagrant/go/src/github.com/cilium/cilium'
[2020-05-20T21:36:16.370Z]   	 
[2020-05-20T21:36:16.370Z]   Stderr:
[2020-05-20T21:36:16.370Z]    	 make: *** [tests-privileged] Error 1
```

**Comment by aanm** [comment]
test-me-please

**Review by jrajahalme — APPROVED** [review]
Nice for a change that all the commits were independent of each other :-)

**Comment by aanm** [comment]
k8s 1.18 hit https://github.com/cilium/cilium/issues/11582 https://jenkins.cilium.io/job/Cilium-PR-K8s-newest-kernel-4.9/312/
4.9 hit https://github.com/cilium/cilium/issues/11076 https://jenkins.cilium.io/job/Cilium-PR-Runtime-4.9/322/

travis hit https://github.com/cilium/cilium/issues/11560
